libcxx/include/__atomic/atomic_base.h

```diff
@@ -27,6 +27,20 @@
 
 _LIBCPP_BEGIN_NAMESPACE_STD
 
+#if _LIBCPP_STD_VER >= 20
+template <class _Tp, bool __lock_free>
+inline constexpr bool __deprecated_if_not_always_lock_free = true;
+
+template <class _Tp>
+[[deprecated("volatile atomic operations are deprecated when std::atomic<T>::is_always_lock_free is false")]]
+inline constexpr bool __deprecated_if_not_always_lock_free<_Tp, false> = true;
+#  define _LIBCPP_DEPRECATED_NOT_ALWAYS_LOCK_FREE(_Tp, __is_always_lock_free)                                          \
+    static_assert(__deprecated_if_not_always_lock_free<_Tp, __is_always_lock_free>)
+#else
+#  define _LIBCPP_DEPRECATED_NOT_ALWAYS_LOCK_FREE(_Tp, __is_always_lock_free)                                          \
+    {}
```

Comment:
This needs to be a statement and require a semi-colon.
```suggestion
    do { } while(1)
```

Reply:
is `do { } while(1)` a no-op in C++? I thought it was UB.

Reply:
> I thought it was UB.

It's definitely not now per [P2809R3](https://wg21.link/p2809r3).

However..., shouldn't this be `do { } while(0)`?